lms/static/scripts/frontend_apps/components/dashboard/test/CourseActivity-test.js

```diff
@@ -203,27 +218,43 @@ describe('CourseActivity', () => {
     });
   });
 
-  [12, 35, 1, 500].forEach(id => {
-    it('builds expected href for row confirmation', () => {
-      const wrapper = createComponent();
-      const href = wrapper
+  context('when building row confirmation links', () => {
+    function getLinkForID(wrapper, id) {
+      return wrapper
         .find('OrderableActivityTable')
         .props()
         .navigateOnConfirmRow({ id });
+    }
 
-      assert.equal(href, `/assignments/${id}`);
+    [12, 35, 1, 500].forEach(id => {
+      it('builds expected link for row confirmation', () => {
+        const wrapper = createComponent();
+        assert.equal(getLinkForID(wrapper, id), `/assignments/${id}`);
+      });
     });
-  });
 
-  context('when filters are set', () => {
-    function setCurrentURL(url) {
-      history.replaceState(null, '', url);
-    }
+    [
+      {
+        prop: 'students',
+        arg: ['123', '456'],
+        expectedLink: '/assignments/123?student_id=123&student_id=456',
+      },
+      {
+        prop: 'assignments',
+        arg: ['999'],
+        expectedLink: '/assignments/123', // Selected assignments are not propagated
+      },
+    ].forEach(({ prop, arg, expectedLink }) => {
+      it('takes filters into consideration', () => {
```

Comment:
```suggestion
      it('takes filters into consideration', () => {
```

See comment above about test name.